src/odemis/util/img.py

```diff
@@ -435,7 +435,7 @@ def ensure2DImage(data):
     if len(d.shape) < 2:
         d.shape = (1,) * (2 - len(d.shape)) + d.shape
     elif len(d.shape) > 2:
-        d.shape = d.shape[-2:]  # raise ValueError if it will not work
+        d.shape = d.shape[-2:]  # raise ValueError if it will not work  # TODO will fail if image is already RGB but does not have the MD
```

Comment:
Actually that function was not really meant to be called with RGB data. You could try to extend it to detect MD_DIMS="YXC" and len(C-dim) == 3 or 4.
Do you need that?

Reply:
Don't need it as I now add MD_DIMS to my data, but when I did not have this, it was causing an error. So we can leave it.